vllm/entrypoints/openai/serving_completion.py

```diff
@@ -548,19 +564,22 @@ def request_output_to_completion_response(
             total_tokens=num_prompt_tokens + num_generated_tokens,
         )
 
-        if self.enable_prompt_tokens_details and final_res.num_cached_tokens:
```

Comment:
Note: `final_res` is an internal variable from the for-loop above. If `final_res_batch` is an empty list (`[]`), then `final_res` will not be initialized.